ext/intl/calendar/calendar_ce.h

```diff
@@ -0,0 +1,20 @@
+/*
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | http://www.php.net/license/3_01.txt                                  |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Gustavo Lopes <cataphract@php.net>                          |
+   +----------------------------------------------------------------------+
+ */
+
+#ifndef CALENDAR_CLASS_H
+#define CALENDAR_CLASS_H
+
+extern zend_class_entry *Calendar_ce_ptr, *GregorianCalendar_ce_ptr;
```

Comment:
Why is a separate header for this needed? Aren't these in calendar_class.h already?

Reply:
I got the `error: reference to 'Calendar' is ambiguous` error message, that's the reason. Is there any better alternative to circumvent the issue?

Reply:
Where does the error happen?

Reply:
When compiling `dateformat_create.cpp`, the following error is shown:
```
ext/intl/dateformat/dateformat_helpers.h:35:42: error: reference to 'Calendar' is ambiguous
        const char *func_name, intl_error *err, Calendar*& cal, zend_long& cal_int_type, bool& calendar_owned

ext/intl/dateformat/dateformat_helpers.h:30:12: note: candidate found by name lookup is 'Calendar'
using icu::Calendar;
           ^
ext/intl/dateformat/../calendar/calendar_class.h:26:14: note: candidate found by name lookup is 'Calendar'
typedef void Calendar;
```

As far as I understood, the two `Calendar`s are clashing with each other, but I'm very helpless what to do here.

Reply:
Can you please commit it in a form that gives the error? I suspect you're missing a `#define USE_CALENDAR_POINTER 1`, but I don't really get where.

Reply:
I've committed all the fixes I did so far, with the erroneous define. I think only the issue with the filter default values are left.

Reply:
As expected, this was a missing USE_CALENDAR_POINTER define, fixed in https://github.com/php/php-src/pull/6081/commits/9853823880af5b95755e8b29ef9cd8e9f58ff06d.

Reply:
We can drop the extra header now, right?

Reply:
Yes, I wanted to do so at some point... :)